core/src/test/resources/HistoryServerExpectations/excludeOnFailure_for_stage_expectation.json

```diff
@@ -698,6 +698,28 @@
       "memoryBytesSpilled" : 0,
       "diskBytesSpilled" : 0,
       "isBlacklistedForStage" : true,
+      "peakMemoryMetrics" : {
+        "JVMHeapMemory" : -1,
```

Comment:
The `-1` here is wrong. It is set to -1 on purpose.
```
class ExecutorMetrics private[spark] extends Serializable {
  // Metrics are indexed by ExecutorMetricType.metricToOffset
  private val metrics = new Array[Long](ExecutorMetricType.numMetrics)
  // the first element is initialized to -1, indicating that the values for the array
  // haven't been set yet.
  metrics(0) = -1
```

Reply:
> The `-1` here is wrong. It is set to -1 on purpose.
> 
> ```
> class ExecutorMetrics private[spark] extends Serializable {
>   // Metrics are indexed by ExecutorMetricType.metricToOffset
>   private val metrics = new Array[Long](ExecutorMetricType.numMetrics)
>   // the first element is initialized to -1, indicating that the values for the array
>   // haven't been set yet.
>   metrics(0) = -1
> ```

This is a tricky question, we'd better change a method to judge whether is it set.

Reply:
All `isSet` is called to filter data
![image](https://user-images.githubusercontent.com/46485123/101156470-9b4b5600-3663-11eb-96ed-c1bfed01510c.png)

I didn't think of a better way to deal with the problem. WDYT?

Reply:
Now you don't think about this `-1` right?
We don't need to change JSON files.

Reply:
> After [13f6db2](https://github.com/apache/spark/commit/13f6db2c2e76df99b04ac703f3a803feb31c2328) you don't need to think about this `-1` right?
> We don't need to change JSON files.

Yes, forgot this point, I  have updated code now.